//test exception 11
// test fail allocation.
struct BadList;
typedef struct BadList * BadList;

struct BadList {
       int val;
       BadList next;
};

BadList build(int i) {
	if (i == 0) return NULL;
	BadList curr = alloc(struct BadList);
	curr->val = i;
	curr->next = build(i-1);
	return curr;
}

int getNth(BadList curr, int i) {
	if (i <= 0) return curr->val;
	return getNth(curr->next, i-1);
}

int main () {
    struct BadList [] haha = alloc_array(struct BadList, 0xffffffff);

    return haha[0].val;
}